src/main/app/forms/validation/validators/allFieldsArePopulated.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator'
+
+@ValidatorConstraint()
+export class AllFieldsArePopulatedConstraint implements ValidatorConstraintInterface {
+  validate (value: any, args?: ValidationArguments) {
+    return args.constraints.every(field => args.object[field] && !!args.object[field].trim())
```

Comment:
its not technically needed, as line 1 will always be undefined when it is hidden, I added the trim to make it more generic.
Do you think I should remove it?

Reply:
Yes. Removing it would also make the documentation more accurate:

> Verifies if all of the fields in given object are populated with "truthy" value.

Reply:
I think I'd extract the body of this function into another function which checks if the argument is a string and only then calls `trim`.

Reply:
will go with @geek-soft 's approach, its not needed now and can be added later if needed

Reply:
I mean the arrow function body.